src/Common/SimulationIC.hpp

```diff
@@ -190,11 +183,15 @@ void Simulation<ndim>::GenerateIC(void)
   }
   //-----------------------------------------------------------------------------------------------
 
+
+  // Finally, generate the initial conditions
+  icGenerator->Generate();
+
   // Scale particle data to dimensionless code units if required
   if (rescale_particle_data) ConvertToCodeUnits();
 
   // Check that the initial conditions are valid
-  icGenerator.CheckInitialConditions();
+  icGenerator->CheckInitialConditions();
 
```

Comment:
I don't like too much carrying around the pointer to the initial condition object for the whole duration of the simulation to be honest. Is it needed only in RegulariseParticleDistribution? If yes, can't we find another way of letting the pointer get there?